src/KafkaNET.Library/ZooKeeperIntegration/Listeners/ZKRebalancerListener.cs

```diff
@@ -496,34 +496,19 @@ private void AddPartitionTopicInfo(ZKGroupTopicDirs topicDirs, string partition,
             }
             var leader = leaderOpt.Value;
             var znode = topicDirs.ConsumerOffsetDir + "/" + partition;
-            var offsetString = this.zkClient.ReadData<string>(znode, true);
+
+            var offsetCommitedString = this.zkClient.ReadData<string>(znode, true);
 
             //if first time starting a consumer, set the initial offset based on the config
-            long offset = 0;
-            long offsetCommited = 0;
-            if (offsetString == null)
-            {
-                switch (config.AutoOffsetReset)
```

Comment:
@ducas The auto offset reset logic was redundant here.
If we don't find an offset (i.e. it is the first run for a consumer group), I am setting it "long offset = -1" (not 0) now.
Setting the offset to -1 here, kicks in the same auto offset reset logic in the FetcherRunnable.cs later on (switch case for Offset Out Of Bounds exception).
Java client api has a similar approach too.